Description: Build Gradle using libraries available in Debian
Author: Miguel Landaeta <miguel@miguel.cc>
Author: Damien Raude-Morvan <drazzib@debian.org
Forwarded: not-needed
Last-Update: 2012-08-18

--- a/build.gradle
+++ b/build.gradle
@@ -17,6 +17,7 @@
 import org.gradle.build.Install
 import org.gradle.build.BuildTypes
 import org.gradle.build.TestReportAggregator
+import org.apache.ivy.plugins.resolver.FileSystemResolver
 
 defaultTasks 'assemble'
 apply plugin: 'java-base'
@@ -102,7 +103,11 @@
     group = 'org.gradle'
 
     repositories {
-        maven { url 'http://repo.gradle.org/gradle/libs' }
+    add(new FileSystemResolver()) {
+        name = 'repo'
+        addArtifactPattern "/usr/share/java/[module]-[revision].[ext]"
+        addArtifactPattern "/usr/share/java/[module].[ext]"
+    }
     }
 }
 
@@ -187,4 +192,4 @@
     installDirPropertyName = 'gradle_installPath'
 }
 
-apply from: "gradle/intTestImage.gradle"
\ No newline at end of file
+apply from: "gradle/intTestImage.gradle"
--- a/buildSrc/build.gradle
+++ b/buildSrc/build.gradle
@@ -13,6 +13,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+import org.apache.ivy.plugins.resolver.FileSystemResolver
 
 apply plugin: 'groovy'
 apply plugin: 'checkstyle'
@@ -21,8 +22,14 @@
 apply plugin: 'eclipse'
 
 repositories {
-    maven { url 'http://repo.gradle.org/gradle/libs' }
-    mavenCentral()
+    //maven { url 'http://repo.gradle.org/gradle/libs' }
+    //mavenCentral()
+    //maven { url '/usr/share/maven-repo/' }
+    add(new FileSystemResolver()) {
+        name = 'repo'
+        addArtifactPattern "/usr/share/java/[module]-[revision].[ext]"
+        addArtifactPattern "/usr/share/java/[module].[ext]"
+    }
 }
 
 dependencies {
@@ -30,7 +37,7 @@
     compile 'com.google.guava:guava:11.0.2@jar'
     compile 'commons-lang:commons-lang:2.6@jar'
     groovy localGroovy()
-    testCompile 'junit:junit:4.10@jar'
+    testCompile 'junit:junit4:4.x@jar'
     testCompile 'org.spockframework:spock-core:0.6-groovy-1.8@jar', 'cglib:cglib-nodep:2.2', 'org.objenesis:objenesis:1.2'
 
     compile "org.pegdown:pegdown:1.1.0"
--- a/gradle/dependencies.gradle
+++ b/gradle/dependencies.gradle
@@ -25,16 +25,16 @@
     dependency 'org.apache.ant:ant-launcher:1.8.4@jar'
 }
 
-libraries.asm =  'org.ow2.asm:asm-all:4.0@jar'
+libraries.asm =  'asm:asm4-all:4.1@jar'
 libraries.commons_cli = 'commons-cli:commons-cli:1.2@jar'
 libraries.commons_io = dependencies.module(versions.commons_io)
 libraries.commons_lang = 'commons-lang:commons-lang:2.6@jar'
-libraries.commons_collections = 'commons-collections:commons-collections:3.2.1@jar'
+libraries.commons_collections = 'commons-collections:commons-collections3:3.2.1@jar'
 libraries.ivy = dependencies.module('org.apache.ivy:ivy:2.2.0'){
     dependency "com.jcraft:jsch:0.1.46"
 }
-libraries.jcip = "net.jcip:jcip-annotations:1.0@jar"
-libraries.inject = dependencies.module('javax.inject:javax.inject:1')
+libraries.jcip = "net.jcip:jcip:1.0@jar"
+libraries.inject = dependencies.module('javax.inject:atinject-jsr330-api:1.0')
 
 // Logging
 libraries.slf4j_api = 'org.slf4j:slf4j-api:1.6.6@jar'
@@ -54,7 +54,7 @@
 }
 
 // Jetty
-libraries.servlet_api = "org.mortbay.jetty:servlet-api:2.5-20081211@jar"
+libraries.servlet_api = "javax.servlet:servlet-api:2.5@jar"
 libraries.jetty_util = dependencies.module("org.mortbay.jetty:jetty-util:6.1.25") {
     dependency libraries.slf4j_api
     dependency libraries.servlet_api
@@ -84,23 +84,25 @@
         jsr305: 'com.google.code.findbugs:jsr305:1.3.9',
         groovy: 'org.codehaus.groovy:groovy-all:1.8.6@jar',
         jaxen: 'jaxen:jaxen:1.1@jar',
-        jcip: "net.jcip:jcip-annotations:1.0",
+        jcip: "net.jcip:jcip:debian",
         jna: 'net.java.dev.jna:jna:3.2.7@jar',
-        junit: 'junit:junit:4.10',
+        junit: 'junit:junit4:4.10',
         xmlunit: 'xmlunit:xmlunit:1.3',
-        nekohtml: 'net.sourceforge.nekohtml:nekohtml:1.9.14',
+        nekohtml: dependencies.module('net.sourceforge.nekohtml:nekohtml:1.9.14') {
+		dependency "xerces:xercesImpl:debian@jar"
+	},
         xbean: 'org.apache.xbean:xbean-reflect:3.4@jar', //required by maven3 classes
         nativePlatform: 'net.rubygrapefruit:native-platform:0.2'
 ]
 
-libraries.maven3 = dependencies.module("org.apache.maven:maven-core:3.0.4") {
-    dependency "org.apache.maven:maven-settings:3.0.4@jar"
-    dependency "org.apache.maven:maven-settings-builder:3.0.4@jar"
+libraries.maven3 = dependencies.module("org.apache.maven:maven-core:3.x") {
+    dependency "org.apache.maven:maven-settings:3.x@jar"
+    dependency "org.apache.maven:maven-settings-builder:3.x@jar"
 
     //plexus:
-    dependency "org.codehaus.plexus:plexus-utils:2.0.6@jar"
+    dependency "org.codehaus.plexus:plexus-utils2:debian@jar"
     dependency "org.codehaus.plexus:plexus-interpolation:1.14@jar"
-    dependency "org.codehaus.plexus:plexus-component-annotations:1.5.5@jar"
+    dependency "org.codehaus.plexus:plexus-component-annotations-1.5:debian@jar"
     dependency "org.codehaus.plexus:plexus-container-default:1.5.5@jar"
     dependency "org.codehaus.plexus:plexus-classworlds:2.4@jar"
 
@@ -109,17 +111,17 @@
     dependency "org.sonatype.plexus:plexus-sec-dispatcher:1.3@jar"
 
     //core:
-    dependency "org.apache.maven:maven-core:3.0.4@jar"
-    dependency "org.apache.maven:maven-model-builder:3.0.4@jar"
-    dependency "org.apache.maven:maven-model:3.0.4@jar"
+    dependency "org.apache.maven:maven-core:3.x@jar"
+    dependency "org.apache.maven:maven-model-builder:3.x@jar"
+    dependency "org.apache.maven:maven-model:3.x@jar"
 
     //somewhat core:
-    dependency "org.apache.maven:maven-artifact:3.0.4@jar"
-    dependency "org.apache.maven:maven-compat:3.0.4@jar"
-    dependency "org.apache.maven:maven-repository-metadata:3.0.4@jar"
-    dependency "org.apache.maven:maven-plugin-api:3.0.4@jar"
-    dependency "org.apache.maven:maven-aether-provider:3.0.4@jar"
-    dependency "org.apache.maven.wagon:wagon-provider-api:2.2@jar"
+    dependency "org.apache.maven:maven-artifact:3.x@jar"
+    dependency "org.apache.maven:maven-compat:3.x@jar"
+    dependency "org.apache.maven:maven-repository-metadata:3.x@jar"
+    dependency "org.apache.maven:maven-plugin-api:3.x@jar"
+    dependency "org.apache.maven:maven-aether-provider:3.x@jar"
+    dependency "org.apache.maven.wagon:wagon-provider-api:2.x@jar"
 
     //eather:
     dependency "org.sonatype.aether:aether-api:1.13.1@jar"
--- a/subprojects/core/core.gradle
+++ b/subprojects/core/core.gradle
@@ -57,7 +57,7 @@
 
     integTestCompile project(":internalIntegTesting")
 
-    reports 'css3-pie:css3-pie:1.0beta3'
+    //reports 'css3-pie:css3-pie:1.0beta3'
 }
 
 useTestFixtures()
--- a/subprojects/core/src/main/groovy/org/gradle/api/internal/classpath/DefaultModuleRegistry.java
+++ b/subprojects/core/src/main/groovy/org/gradle/api/internal/classpath/DefaultModuleRegistry.java
@@ -223,15 +223,18 @@
     }
 
     private File findExternalJar(String name) {
-        Pattern pattern = Pattern.compile(Pattern.quote(name) + "-\\d.+\\.jar");
+        Pattern patternVersion = Pattern.compile(Pattern.quote(name) + "-\\d.+\\.jar");
+        Pattern pattern = Pattern.compile(Pattern.quote(name) + ".jar");
         for (File file : classpath) {
-            if (pattern.matcher(file.getName()).matches()) {
+            if (patternVersion.matcher(file.getName()).matches()
+            || pattern.matcher(file.getName()).matches()) {
                 return file;
             }
         }
         for (File libDir : libDirs) {
             for (File file : libDir.listFiles()) {
-                if (pattern.matcher(file.getName()).matches()) {
+                if (patternVersion.matcher(file.getName()).matches()
+                || pattern.matcher(file.getName()).matches()) {
                     return file;
                 }
             }
--- a/subprojects/core/src/main/groovy/org/gradle/profile/ProfileReportRenderer.java
+++ b/subprojects/core/src/main/groovy/org/gradle/profile/ProfileReportRenderer.java
@@ -30,7 +30,7 @@
         HtmlReportRenderer renderer = new HtmlReportRenderer();
         renderer.requireResource(getClass().getResource("/org/gradle/reporting/base-style.css"));
         renderer.requireResource(getClass().getResource("/org/gradle/reporting/report.js"));
-        renderer.requireResource(getClass().getResource("/org/gradle/reporting/css3-pie-1.0beta3.htc"));
+        //renderer.requireResource(getClass().getResource("/org/gradle/reporting/css3-pie-1.0beta3.htc"));
         renderer.requireResource(getClass().getResource("style.css"));
         renderer.renderer(new ProfilePageRenderer()).writeTo(buildProfile, file);
     }
--- a/subprojects/core/src/main/resources/org/gradle/reporting/base-style.css
+++ b/subprojects/core/src/main/resources/org/gradle/reporting/base-style.css
@@ -68,7 +68,7 @@
     margin-right: 25px;
     border: solid 1px #d4d4d4;
     background-color: #f0f0f0;
-    behavior: url(css3-pie-1.0beta3.htc);
+    /*behavior: url(css3-pie-1.0beta3.htc);*/
 }
 
 ul.tabLinks li:hover {
--- a/subprojects/javascript/javascript.gradle
+++ b/subprojects/javascript/javascript.gradle
@@ -17,7 +17,7 @@
 dependencies {
     groovy libraries.groovy
 
-    compile "org.mozilla:rhino:1.7R3"
+    compile "org.mozilla:js:1.7R3"
     compile "com.google.code.gson:gson:2.2.1" // used by JsHint
     compile "org.simpleframework:simple:4.1.21" // used by http package in envjs
     compile project(':core'), project(":plugins")
--- a/subprojects/jetty/jetty.gradle
+++ b/subprojects/jetty/jetty.gradle
@@ -23,22 +23,22 @@
     compile libraries.jetty
     compile libraries.jetty_util
     compile libraries.servlet_api
-    compile module("org.mortbay.jetty:jetty-plus:6.1.25") {
+    compile module("org.mortbay.jetty:jetty-plus:6.x") {
         dependency libraries.jetty
-        module("org.mortbay.jetty:jetty-naming:6.1.25") {
+        module("org.mortbay.jetty:jetty-naming:6.x") {
             dependency libraries.jetty
         }
     }
 
-    runtime module("org.mortbay.jetty:jsp-2.1:6.1.14") {
-        dependency "org.eclipse.jdt:core:3.1.1@jar"
-        dependency "org.mortbay.jetty:jsp-api-2.1:6.1.14@jar"
+    runtime module("org.mortbay.jetty:jsp-api:2.1") {
+        dependency "org.eclipse.jdt:eclipse-ecj:3.1.1@jar"
+        dependency "org.mortbay.jetty:jsp-api-2.1:6.x@jar"
         dependency libraries.jetty_util
         dependency libraries.servlet_api
     }
 
-    runtime "org.mortbay.jetty:jetty-annotations:6.1.25@jar"
-    runtime "org.apache.geronimo.specs:geronimo-annotation_1.0_spec:1.0@jar"
+    runtime "org.mortbay.jetty:jetty-annotations:6.x@jar"
+    //runtime "org.apache.geronimo.specs:geronimo-annotation_1.0_spec:1.0@jar"
 }
 
-useTestFixtures()
\ No newline at end of file
+useTestFixtures()
--- a/subprojects/maven/maven.gradle
+++ b/subprojects/maven/maven.gradle
@@ -25,7 +25,7 @@
     compile libraries.maven_ant_tasks
     compile "org.sonatype.pmaven:pmaven-common:0.8-20100325@jar"
     compile "org.sonatype.pmaven:pmaven-groovy:0.8-20100325@jar"
-    compile "org.codehaus.plexus:plexus-component-annotations:1.5.2@jar"
+    compile "org.codehaus.plexus:plexus-component-annotations-1.5:debian@jar"
 }
 
 useTestFixtures()
--- a/subprojects/osgi/osgi.gradle
+++ b/subprojects/osgi/osgi.gradle
@@ -21,7 +21,7 @@
     compile project(':plugins')
     compile libraries.slf4j_api
 
-    compile module('biz.aQute:bndlib:1.50.0')
+    compile module('biz.aQute:bndlib:debian')
 }
 
 useTestFixtures()
--- a/subprojects/plugins/plugins.gradle
+++ b/subprojects/plugins/plugins.gradle
@@ -38,7 +38,7 @@
     compile libraries.commons_lang
     compile libraries.junit
     compile libraries.slf4j_api
-    compile 'org.testng:testng:6.3.1'
+    compile 'org.testng:testng:5.11@jar'
 
     provided files(jvm.toolsJar) // for SunJavaCompiler
 
--- a/subprojects/plugins/src/main/groovy/org/gradle/api/internal/tasks/testing/junit/report/DefaultTestReport.java
+++ b/subprojects/plugins/src/main/groovy/org/gradle/api/internal/tasks/testing/junit/report/DefaultTestReport.java
@@ -36,7 +36,7 @@
     public DefaultTestReport() {
         htmlRenderer.requireResource(getClass().getResource("/org/gradle/reporting/report.js"));
         htmlRenderer.requireResource(getClass().getResource("/org/gradle/reporting/base-style.css"));
-        htmlRenderer.requireResource(getClass().getResource("/org/gradle/reporting/css3-pie-1.0beta3.htc"));
+        //htmlRenderer.requireResource(getClass().getResource("/org/gradle/reporting/css3-pie-1.0beta3.htc"));
         htmlRenderer.requireResource(getClass().getResource("style.css"));
     }
 
--- a/subprojects/plugins/src/main/resources/org/gradle/api/internal/tasks/testing/junit/report/style.css
+++ b/subprojects/plugins/src/main/resources/org/gradle/api/internal/tasks/testing/junit/report/style.css
@@ -41,7 +41,7 @@
     border: solid 2px #d0d0d0;
     -moz-border-radius: 10px;
     border-radius: 10px;
-    behavior: url(css3-pie-1.0beta3.htc);
+    /*behavior: url(css3-pie-1.0beta3.htc);*/
 }
 
 #successRate {
--- a/subprojects/signing/signing.gradle
+++ b/subprojects/signing/signing.gradle
@@ -21,9 +21,9 @@
     compile project(":plugins")
     compile project(":maven")
 
-    compile module("org.bouncycastle:bcpg-jdk15:1.46") {
-        dependency "org.bouncycastle:bcprov-jdk15:1.46@jar"
+    compile module("org.bouncycastle:bcpg:debian") {
+        dependency "org.bouncycastle:bcprov:debian@jar"
     }
 }
 
-useTestFixtures()
\ No newline at end of file
+useTestFixtures()
