Description: Plexus compatibility
Author: Miguel Landaeta <miguel@miguel.cc>
Author: Damien Raude-Morvan <drazzib@debian.org
Forwarded: no
Last-Update: 2012-08-18

--- a/subprojects/core-impl/src/main/groovy/org/gradle/api/internal/artifacts/PlexusLoggerAdapter.java
+++ b/subprojects/core-impl/src/main/groovy/org/gradle/api/internal/artifacts/PlexusLoggerAdapter.java
@@ -98,4 +98,8 @@
     public String getName() {
         return logger.getName();
     }
+
+    public void setThreshold(int level) {
+        throw new UnsupportedOperationException("setThreshold");
+    }
 }
