Description: Remove build timestamp from the version string
Author: Miguel Landaeta <miguel@miguel.cc>
Forwarded: not-needed
Last-Update: 2010-11-23

--- gradle-0.9~rc3.orig/buildSrc/src/main/groovy/org/gradle/build/Version.groovy
+++ gradle-0.9~rc3/buildSrc/src/main/groovy/org/gradle/build/Version.groovy
@@ -31,7 +31,7 @@ class Version {
 
         project.gradle.taskGraph.whenReady {graph ->
             if (!graph.hasTask(':releaseVersion')) {
-                this.versionNumber += "-" + getTimestamp()
+                //this.versionNumber += "-" + getTimestamp()
                 release = false
             } else {
                 release = true
