Author: Damien Raude-Morvan <drazzib@debian.org>
Forwarded: not-needed
Last-Update: 2012-08-18

--- a/build.gradle
+++ b/build.gradle
@@ -224,7 +224,7 @@
     testResultsDir = file("${buildDir}/test-results")
     projects = subprojects
 }
-
+/*
 task determineCommitId {
     ext.commitId = null
 
@@ -269,14 +269,15 @@
         }
     }
 }
+*/
 
-task createBuildReceipt(dependsOn: determineCommitId) {
+task createBuildReceipt {
     ext.receiptFile = file("$buildDir/build-receipt.properties")
     outputs.file receiptFile
     outputs.upToDateWhen { false }
     doLast {
         def data = [
-            commitId:  determineCommitId.commitId,
+            commitId:  "UNKNOWN",
             versionNumber: version,
             buildTimestamp: buildTimestamp,
             username: System.properties["user.name"],
