Author: Damien Raude-Morvan <drazzib@debian.org>
Forwarded: not-needed
Last-Update: 2012-08-18

--- a/gradle/buildReceipt.gradle
+++ b/gradle/buildReceipt.gradle
@@ -32,7 +32,7 @@
     }
 }
 
-task determineCommitId {
+/*task determineCommitId {
     ext.commitId = null
 
     doLast {
@@ -93,9 +93,9 @@
             throw new InvalidUserDataException("Could not determine commit id")
         }
     }
-}
+}*/
 
-task createBuildReceipt(dependsOn: determineCommitId) {
+task createBuildReceipt() {
     ext.receiptFile = file("$buildDir/$buildReceiptFileName")
     outputs.file receiptFile
     outputs.upToDateWhen { false }
@@ -107,7 +107,7 @@
             hostName = "unknown"
         }
         def data = [
-                commitId:  determineCommitId.commitId,
+                commitId: "UNKNOWN",
                 versionNumber: version,
                 buildTimestamp: buildTimestamp,
                 username: System.properties["user.name"],
