Author: Damien Raude-Morvan <drazzib@debian.org>
Forwarded: not-needed
Last-Update: 2012-08-18

--- a/subprojects/core-impl/core-impl.gradle
+++ b/subprojects/core-impl/core-impl.gradle
@@ -58,7 +58,7 @@
     //I'm removing the components.xml file from the sec-dispatcher jar.
     //This file contains only single component so I think we can remove it.
     task.doLast {
-        def plexusSec = "$outputDir/jarjar-plexus-sec-dispatcher-1.3.jar"
+        def plexusSec = "$outputDir/jarjar-plexus-sec-dispatcher.jar"
         def plexusSecNoComps = "$plexusSec-noComps"
         ant {
             zip(destfile: plexusSecNoComps, update: true) {
