--- a/subprojects/core-impl/src/main/groovy/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/NoOpRepositoryCacheManager.java
+++ b/subprojects/core-impl/src/main/groovy/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/NoOpRepositoryCacheManager.java
@@ -55,12 +55,17 @@
         return null;
     }
 
+    public org.apache.ivy.core.report.ArtifactDownloadReport downloadRepositoryResource(org.apache.ivy.plugins.repository.Resource resource,
+           java.lang.String name, java.lang.String type, java.lang.String extension, org.apache.ivy.core.cache.CacheResourceOptions options, org.apache.ivy.plugins.repository.Repository repository) {
+         return null;
+    }
+
     public ArtifactDownloadReport download(Artifact artifact, ArtifactResourceResolver resourceResolver, ResourceDownloader resourceDownloader, CacheDownloadOptions options) {
         ArtifactDownloadReport report = new ArtifactDownloadReport(null);
         report.setDownloadStatus(DownloadStatus.NO);
         return report;
     }
-
+    
     public ResolvedModuleRevision cacheModuleDescriptor(DependencyResolver resolver, ResolvedResource orginalMetadataRef, DependencyDescriptor dd, Artifact requestedMetadataArtifact, ResourceDownloader downloader, CacheMetadataOptions options) throws ParseException {
         return null;
     }
--- a/subprojects/core-impl/src/main/groovy/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/parser/IvyXmlModuleDescriptorParser.java
+++ b/subprojects/core-impl/src/main/groovy/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/parser/IvyXmlModuleDescriptorParser.java
@@ -318,8 +318,7 @@
             }
 
             DefaultExtendsDescriptor ed = new DefaultExtendsDescriptor(
-                    parent.getModuleRevisionId(),
-                    parent.getResolvedModuleRevisionId(),
+                    parent,
                     attributes.getValue("location"),
                     extendTypes.toArray(new String[extendTypes.size()]));
             getMd().addInheritedDescriptor(ed);
@@ -924,4 +923,4 @@
     public String toString() {
         return "ivy parser";
     }
-}
\ No newline at end of file
+}
--- a/subprojects/core-impl/src/main/groovy/org/gradle/api/internal/artifacts/repositories/cachemanager/AbstractRepositoryCacheManager.java
+++ b/subprojects/core-impl/src/main/groovy/org/gradle/api/internal/artifacts/repositories/cachemanager/AbstractRepositoryCacheManager.java
@@ -63,6 +63,11 @@
         return null;
     }
 
+    public org.apache.ivy.core.report.ArtifactDownloadReport downloadRepositoryResource(org.apache.ivy.plugins.repository.Resource resource,
+           java.lang.String name, java.lang.String type, java.lang.String extension, org.apache.ivy.core.cache.CacheResourceOptions options, org.apache.ivy.plugins.repository.Repository repository) {
+         return null;
+    }
+
     public void originalToCachedModuleDescriptor(DependencyResolver resolver, ResolvedResource originalMetadataRef, Artifact requestedMetadataArtifact, ResolvedModuleRevision rmr, ModuleDescriptorWriter writer) {
     }
 
